File: src/jvmMain/java/ess/papyrus/ParamBool.java
package ess.papyrus;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.nio.ByteBuffer;
import java.util.Objects;

/**
 * An opcode parameter that stores a boolean.
 */
final public class ParamBool extends Parameter {

    public ParamBool(byte val) {
        this.VALUE = val;
    }

    @NotNull
    @Override
    public ParamType getType() {
        return ParamType.BOOLEAN;
    }

    @Override
    public void write(@NotNull ByteBuffer output) {
        this.getType().write(output);
        output.put(this.VALUE);
    }

    @Override
    public int calculateSize() {
        return 2;
    }

    @NotNull
    @Override
    public String toValueString() {
        return Boolean.toString(this.VALUE != 0);
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 41 * hash + Objects.hashCode(this.getType());
        hash = 41 * hash + Byte.hashCode(this.VALUE);
        return hash;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (this == obj) {
            return true;
        } else if (obj == null) {
            return false;
        } else if (getClass() != obj.getClass()) {
            return false;
        }
        final ParamBool other = (ParamBool) obj;
        return this.VALUE == other.VALUE;
    }

    final public byte VALUE;
}

File: src/jvmMain/kotlin/ess/papyrus/ParamBool.kt
package ess.papyrus

import java.nio.ByteBuffer


/**
 * An opcode parameter that stores a boolean.
 */
class ParamBool(val VALUE: Byte) : Parameter() {
    override val type: ParamType
        get() = ParamType.BOOLEAN

    override fun write(output: ByteBuffer?) {
        type.write(output)
        output?.put(VALUE)
    }

    override fun calculateSize(): Int {
        return 2
    }

    override fun toValueString(): String {
        return java.lang.Boolean.toString(VALUE.toInt() != 0)
    }

    override fun hashCode(): Int {
        var hash = 7
        hash = 41 * hash + type.hashCode()
        hash = 41 * hash + VALUE.hashCode()
        return hash
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        } else if (other == null) {
            return false
        } else if (javaClass != other.javaClass) {
            return false
        }
        val other2 = other as ParamBool
        return VALUE == other2.VALUE
    }
}